canonical-web-and-design/maas-ui
# feat(ui): download machine output

## Done

- Fetch machine summary files and storing them in the React context and allow downloading the files.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- View the React logs tab for a commissioned or deployed machine (change /summary to /logs).
- Open the download menu to the right of the sub-tabs.
- Click the YAML and XML items and the files should download.

## Fixes

Fixes: canonical-web-and-design/maas-ui#2426.
Fixes: canonical-web-and-design/maas-squad#2367.
